dango/lending/src/execute.rs

```diff
@@ -0,0 +1,183 @@
+use {
+    crate::{DEBTS, MARKETS},
+    anyhow::{bail, ensure, Ok},
+    dango_types::{
+        account_factory::QueryAccountRequest,
+        bank,
+        config::ACCOUNT_FACTORY_KEY,
+        lending::{ExecuteMsg, InstantiateMsg, Market, MarketUpdates, NAMESPACE},
+    },
+    grug::{Addr, Coin, Coins, Denom, Inner, Message, MutableCtx, Part, Response},
+    std::{collections::BTreeMap, str::FromStr},
+};
+
+#[cfg_attr(not(feature = "library"), grug::export)]
+pub fn instantiate(ctx: MutableCtx, msg: InstantiateMsg) -> anyhow::Result<Response> {
+    for (denom, _updates) in msg.markets {
+        MARKETS.save(ctx.storage, &denom, &Market {})?;
+    }
+
+    Ok(Response::new())
+}
+
+#[cfg_attr(not(feature = "library"), grug::export)]
+pub fn receive(_ctx: MutableCtx) -> anyhow::Result<Response> {
+    // Reject all transfers.
+    bail!("Can't send tokens to this contract. Use the `deposit` message instead.");
+}
```

Comment:
You don't actually need this. If the `receive` entry point isn't implemented, all transfers to it will already fail.

Reply:
See: dd993db1